ibis/expr/types/maps.py

```diff
@@ -486,6 +484,8 @@ def map(
     │ {'b': 3}             │
     └──────────────────────┘
     """
+    if isinstance(keys, MapValue):
+        return keys
```

Comment:
This functionality seems unrelated to the original issue and needs tests. Can you add this feature in a follow up PR?